packages/grafana-schema/src/schema/dashboard/dashboard.gen.ts

```diff
@@ -0,0 +1,216 @@
+// This file is autogenerated. DO NOT EDIT.
+//
+// Derived from the Thema lineage at pkg/coremodel/dashboard/lineage.cue
+
+export interface Dashboard {
+  annotations?: {
+    list: AnnotationQuery[];
+  };
+  description?: string;
+  editable: boolean;
+  fiscalYearStartMonth?: number;
+  gnetId?: string;
+  graphTooltip: DashboardCursorSync;
+  id?: number;
+  links?: DashboardLink[];
+  liveNow?: boolean;
+  panels?: Panel | {
+      type: 'graph';
+    } | {
+      type: 'heatmap';
+    } | {
+      type: 'row';
+      collapsed: boolean;
+      id: number;
+      panels: Panel | {
+          type: 'graph';
+        } | {
+          type: 'heatmap';
+        }[];
+    }[];
```

Comment:
```suggestion
  panels: Panel[];
```

The multiple types is not going to work here...

Reply:
> banging on it until the behavior is what we want is the only real approach we have

Yes -- i think so.  The go struct > typescript process has a nice setup that essentially lets you force anything special using struct tags.  ie: https://github.com/grafana/grafana/blob/main/pkg/services/live/pipeline/config.go#L37

Reply:
To avoid ambiguity - this is a suggested change, but i'm not planning on merging it because that would be changing generated code. Instead, i'm interpreting it as a point for discussion, and not a blocker for merging this PR.

My sense is that it's better to have this generator in and we can iterate (rapidly, now that cuetsy is finally unblocked and we're about to begin codemobbing) on changing the schema and/or cuetsy in order to achieve the desired output. In other words, this PR is more about setting up the workflow for rapid iteration than the exact state of outputs _right now_.

Reply:
We should only merge the typescript when it is both accurate and plausible to use.  We do not want to merge a `dashboards.gen.ts` that is not accurate and conflicts with existing dashboard models.

I don't care what the semantics for making it generate a useful file are

Reply:
we discussed elsewhere, but for transparency - I'll move the generated output to somewhere else such that it won't be interpreted as part of an API surface.

Or - just had the thought - what if we just don't export any of the types?